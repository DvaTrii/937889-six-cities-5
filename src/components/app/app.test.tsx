import * as React from "react";
import * as renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import App from "./app";
import thunk from "redux-thunk";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe(`Render App`, () => {

  const store = mockStore({
    APP: {
      city: `Amsterdam`,
      activeSorter: `Popular`,
      hoveredOfferId: -1,
    },
    USER: {
      authorizationStatus: false,
      userInfo: {email: ``},
    },
    DATA: {
      offers: []
    }
  });

  it(`Render App`, () => {
    const tree = renderer
      .create(
          <Provider store={store}>
            <App />
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
