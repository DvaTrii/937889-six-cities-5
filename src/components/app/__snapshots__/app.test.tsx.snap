// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App Render App 1`] = `
<div
  className="page page--gray page--main"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Sign In
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--index
      page__main--index-empty"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <div
      className="tabs"
    >
      <section
        className="locations container"
      >
        <ul
          className="locations__list tabs__list"
        >
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item "
              href="#"
            >
              <span>
                Paris
              </span>
            </a>
          </li>
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item "
              href="#"
            >
              <span>
                Cologne
              </span>
            </a>
          </li>
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item "
              href="#"
            >
              <span>
                Brussels
              </span>
            </a>
          </li>
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item tabs__item--active"
              href="#"
            >
              <span>
                Amsterdam
              </span>
            </a>
          </li>
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item "
              href="#"
            >
              <span>
                Hamburg
              </span>
            </a>
          </li>
          <li
            className="locations__item"
            onClick={[Function]}
          >
            <a
              className="locations__item-link tabs__item "
              href="#"
            >
              <span>
                Dusseldorf
              </span>
            </a>
          </li>
        </ul>
      </section>
    </div>
    <div
      className="cities"
    >
      <div
        className="cities__places-container cities__places-container--empty container"
      >
        <section
          className="cities__no-places"
        >
          <div
            className="cities__status-wrapper tabs__content"
          >
            <b
              className="cities__status"
            >
              No places to stay available
            </b>
            <p
              className="cities__status-description"
            >
              We could not find any property available at the moment in 
              Amsterdam
            </p>
          </div>
        </section>
        <div
          className="cities__right-section"
        />
      </div>
    </div>
  </main>
</div>
`;
